core/request_builder_test.go

```diff
@@ -225,6 +225,43 @@ func TestURLEncodedForm(t *testing.T) {
 	assert.NotNil(t, request)
 }
 
+func TestBuildWithMultipartFormWithRepeatedKeys(t *testing.T) {
+	var str = "hello"
+	json1 := make(map[string]interface{})
+	json1["name1"] = "test name1"
+
+	json2 := make(map[string]interface{})
+	json2["name2"] = "test name2"
+
+	builder := NewRequestBuilder("POST")
+	_, err := builder.ConstructHTTPURL("test.com", nil, nil)
+	assert.Nil(t, err)
+
+	builder.AddHeader("Content-Type", "Application/json").
+		AddQuery("Version", "2018-22-09").
+		AddFormData("name", "json1.json", "application/json", json1).
+		AddFormData("name", "json2.json", "application/json", json2).
+		AddFormData("hello", "", "text/plain", "Hello GO SDK").
+		AddFormData("hello", "", "text/plain", &str)
+
+	pwd, _ := os.Getwd()
+	var testFile io.ReadCloser
+	testFile, err = os.Open(pwd + "/../resources/test_file.txt")
+	assert.Nil(t, err, "Could not open file")
+	builder.AddFormData("test_file1", "test_file.txt", "application/octet-stream", testFile)
+	builder.AddFormData("test_file2", "test_file.txt", "application/octet-stream", &testFile)
```

Comment:
Did you intend to use duplicate part names here?

Reply:
Once again, I just copied those from the test above. I was only testing "name", but no reason not to make these duplicates since they are in the test.